osmomath/binary_search.go

```diff
@@ -141,6 +141,63 @@ func (e ErrTolerance) CompareBigDec(expected BigDec, actual BigDec) int {
 	return 0
 }
 
+// CompareDec validates if actual is within errTolerance of expected.
+// returns 0 if it is
+// returns 1 if not, and expected > actual.
+// returns -1 if not, and expected < actual
+func (e ErrTolerance) CompareDec(expected Dec, actual Dec) int {
```

Comment:
Would it be possible to somehow make a generic version shared by all of `Dec`, `BigDec`, `Int` etc?

Reply:
yes, I also thought about that one, totally possible to make it, should I make it in this PR?

Reply:
Let's do in the next PR after merge please